.azure-pipelines/build-template.yml

```diff
@@ -186,7 +186,11 @@ jobs:
         make check
         gcovr --version
         find SAI/meta -name "*.gc*" | xargs rm -vf
-        gcovr -r ./ -e ".*/SAI/.*" -e ".+/json.hpp" -e "swss/.+" -e ".*/.libs/.*" -e ".*/debian/.*" --exclude-unreachable-branches -x --xml-pretty -o coverage.xml
+        cp meta/tests-Meta.gc* unittest/meta
+        pushd unittest/meta
+        gcov *.gc*
+        popd
```

Comment:
What's the purpose of the above 4 lines? To include existing code coverage information from `meta/tests-Meta*` files?

Reply:
It's to include code coverage information for `meta/Meta.cpp`. Currently `Meta.cpp` is reported as 0% covered even though `unittest/meta/TestMeta.cpp` covers most of the file. I believe it's to do with how `gcov`/`gcovr` deals with relative file paths. The `.gcda` file with the coverage information for `Meta.cpp` is generated in `meta/` after running the tests in `unittest/meta`, but because the working directory when running the tests is `unittest/meta`, the `.gcda` files have the relative source path as `../../meta`. When `gcov` goes to generate coverage information, it finds the `.gcda` for `Meta.cpp` file in `meta/`, then tries to access the source file at `meta/../../meta` which doesn't exist.